### PR TITLE
allow CDN to be configured

### LINKED ISSUE #1: Allow CDN to be configured

### PR DESCRIPTION
**Pull request by Preisschild**
closes #1

### CODE REVIEW

File: render_swagger.py [hunk 1]
```diff
@@ -30,6 +32,28 @@
...
+    """
+    Provides the actual swagger library used
+    """
+    lib_swagger = {
+        'css': "https://unpkg.com/swagger-ui-dist@3/swagger-ui.css",
+        'js': "https://unpkg.com/swagger-ui-dist@3/swagger-ui-bundle.js"
+    }
+
+    extra_javascript = config.get('extra_javascript', [])
+    extra_css = config.get('extra_css')
```
Comment (bharel) [inline render_swagger.py:46]:
```suggestion
    extra_css = config.get('extra_css', [])
```

File: render_swagger.py [hunk 2]
```diff
@@ -30,6 +32,28 @@
...
+    for lib in extra_javascript:
+        if os.path.basename(lib) == 'swagger-ui-bundle.js':
```
Comment (bharel) [inline render_swagger.py:48]:
```suggestion
        if os.path.basename(urllib.parse.urlparse(lib).path) == 'swagger-ui-bundle.js':
```

File: render_swagger.py [hunk 3]
```diff
@@ -30,6 +32,28 @@
...
+            lib_swagger['js'] = lib
+
+    for css in extra_css:
+        if os.path.basename(css) == 'swagger-ui.css':
```
Comment (bharel) [inline render_swagger.py:52]:
```suggestion
        if os.path.basename(urllib.parse.urlparse(css).path) == 'swagger-ui.css':
```

File: render_swagger.py [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import os
```
Comment (bharel) [inline render_swagger.py:1]:
```suggestion
import os
import urllib.parse
```

File: render_swagger.py [hunk 5]
```diff
@@ -30,6 +32,28 @@
...
+
+    extra_javascript = config.get('extra_javascript', [])
+    extra_css = config.get('extra_css')
+    for lib in extra_javascript:
+        if os.path.basename(lib) == 'swagger-ui-bundle.js':
+            lib_swagger['js'] = lib
+
+    for css in extra_css:
+        if os.path.basename(css) == 'swagger-ui.css':
+            lib_swagger['css'] = css
```
Comment (bharel) [inline render_swagger.py:53]:
```suggestion
            lib_swagger['css'] = css
            break
```

File: render_swagger.py [hunk 6]
```diff
@@ -30,6 +32,28 @@
...
+    lib_swagger = {
+        'css': "https://unpkg.com/swagger-ui-dist@3/swagger-ui.css",
+        'js': "https://unpkg.com/swagger-ui-dist@3/swagger-ui-bundle.js"
+    }
+
+    extra_javascript = config.get('extra_javascript', [])
+    extra_css = config.get('extra_css')
+    for lib in extra_javascript:
+        if os.path.basename(lib) == 'swagger-ui-bundle.js':
+            lib_swagger['js'] = lib
```
Comment (bharel) [inline render_swagger.py:49]:
```suggestion
            lib_swagger['js'] = lib
            break
```

**Review by bharel — CHANGES_REQUESTED** [review]
Let's play nice with the URLs 🙂 
You can add all suggestions in a batch if you wish.

**Comment by Preisschild** [comment]
Great suggestions.

Thanks for the review!

**Review by bharel — APPROVED** [review]

**Comment by bharel** [comment]
@Preisschild I'll update the version to 0.0.2 so you'll use it if you wish.

We still need some tests before we go stable. To be honest, I'm not sure how to test this together with MKDocs output.
Probably won't be too easy.

**Comment by Preisschild** [comment]
We could use bats to create cli level end to end tests.

The [mkdocs-monorepo-plugin](https://github.com/backstage/mkdocs-monorepo-plugin) does this and it seems to work fairly well

**Pull request merged**